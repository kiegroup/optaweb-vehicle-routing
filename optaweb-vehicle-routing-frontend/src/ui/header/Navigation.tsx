/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Nav, NavItem, NavList, NavVariants } from '@patternfly/react-core';
import * as React from 'react';
import { Link } from 'react-router-dom';

const Navigation: React.FC = () => {
  return (
    <Nav aria-label="Nav">
      <NavList variant={NavVariants.horizontal}>
        <NavItem itemId={'vehicles'}>
          <Link to="/vehicles">Vehicles</Link>
        </NavItem>
        <NavItem itemId={'visits'}>
          <Link to="/visits">Visits</Link>
        </NavItem>
        <NavItem itemId={'route'}>
          <Link to="/route">Route</Link>
        </NavItem>
        <NavItem itemId={'demo'}>
          <Link to="/demo">Demo</Link>
        </NavItem>
      </NavList>
    </Nav>
  );
};

export default Navigation;
