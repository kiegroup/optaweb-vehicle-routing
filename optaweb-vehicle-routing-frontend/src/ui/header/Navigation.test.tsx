import { NavItem } from '@patternfly/react-core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Location } from 'history';
import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Navigation } from './Navigation';

describe('Navigation', () => {
  it('should activate a navigation link matching the current path', () => {
    const props: RouteComponentProps = {
      location: {
        pathname: '/visits',
      } as Location<unknown>,
    } as RouteComponentProps;

    const visitsId = '/visits';

    const navigation = shallow(<Navigation {...props} />);
    expect(toJson(navigation)).toMatchSnapshot();

    // NavItem matching the path should be active
    const navItems = navigation.find(NavItem).filterWhere((navItem) => navItem.props().itemId === visitsId);
    expect(navItems).toHaveLength(1);
    expect(navItems.at(0).props().isActive).toEqual(true);

    // Other NavItems should be inactive
    navigation.find(NavItem).filterWhere((navItem) => navItem.props().itemId !== visitsId).forEach(
      (navItem) => expect(navItem.props().isActive).toEqual(false),
    );
  });
});
