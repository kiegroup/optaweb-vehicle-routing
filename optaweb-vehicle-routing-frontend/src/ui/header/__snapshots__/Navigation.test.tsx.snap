// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should activate a navigation link matching the current path 1`] = `
<Nav
  aria-label="Nav"
  className=""
  onSelect={[Function]}
  onToggle={[Function]}
>
  <NavList
    className=""
    variant="horizontal"
  >
    <NavItem
      className=""
      groupId={null}
      isActive={false}
      itemId="vehicles"
      key="vehicles"
      onClick={null}
      preventDefault={false}
      to=""
    >
      <Link
        to="/vehicles"
      >
        Vehicles
      </Link>
    </NavItem>
    <NavItem
      className=""
      groupId={null}
      isActive={true}
      itemId="visits"
      key="visits"
      onClick={null}
      preventDefault={false}
      to=""
    >
      <Link
        to="/visits"
      >
        Visits
      </Link>
    </NavItem>
    <NavItem
      className=""
      groupId={null}
      isActive={false}
      itemId="route"
      key="route"
      onClick={null}
      preventDefault={false}
      to=""
    >
      <Link
        to="/route"
      >
        Route
      </Link>
    </NavItem>
    <NavItem
      className=""
      groupId={null}
      isActive={false}
      itemId="demo"
      key="demo"
      onClick={null}
      preventDefault={false}
      to=""
    >
      <Link
        to="/demo"
      >
        Demo
      </Link>
    </NavItem>
  </NavList>
</Nav>
`;
